torch/distributed/rpc/options.py

```diff
@@ -0,0 +1,77 @@
+from . import _TensorPipeRpcBackendOptionsBase
+from . import constants as rpc_contants
+
+import torch
+
+from typing import Dict, List
+
+
+class TensorPipeRpcBackendOptions(_TensorPipeRpcBackendOptionsBase):
+    r"""
+    The backend options for
+    :class:`~torch.distributed.rpc.TensorPipeAgent`, derived from
+    :class:`~torch.distributed.rpc.RpcBackendOptions`.
+
+    Arguments:
+        num_worker_threads (int, optional): The number of threads in the
+            thread-pool used by
+            :class:`~torch.distributed.rpc.TensorPipeAgent` to execute
+            requests (default: 16).
+        rpc_timeout (float, optional): The default timeout, in seconds,
+            for RPC requests (default: 60 seconds). If the RPC has not
+            completed in this timeframe, an exception indicating so will
+            be raised. Callers can override this timeout for individual
+            RPCs in :meth:`~torch.distributed.rpc.rpc_sync` and
+            :meth:`~torch.distributed.rpc.rpc_async` if necessary.
+        init_method (str, optional): The URL to initialize the distributed
+            store used for rendezvous. It takes any value accepted for the
+            same argument of :meth:`~torch.distributed.init_process_group`
+            (default: ``env://``).
+    """
+    def __init__(
+        self,
+        num_worker_threads: int = rpc_contants.DEFAULT_NUM_WORKER_THREADS,
+        _transports: List = None,
+        _channels: List = None,
+        rpc_timeout: float = rpc_contants.DEFAULT_RPC_TIMEOUT_SEC,
+        init_method: str = rpc_contants.DEFAULT_INIT_METHOD
+    ):
+        super().__init__(
+            num_worker_threads,
+            _transports,
+            _channels,
+            rpc_timeout,
+            init_method
+        )
+
+    def set_device_map(self, to: str, device_map: Dict):
+        r"""
+        Set device mapping between each RPC caller and callee pair. This
+        function can be called multiple times to incrementally add
+        device placement configurations.
+
+        Arguments:
+            worker_name (str): Callee name.
+            device_map (Dict of int, str, or torch.device): Device placement
```

Comment:
Does the fact that this accepts a variety of types (int, str, torch.device) cause any problems to TorchScript?

Reply:
@wanchaol can `Any` type in TorchScript handle this? BTW, how does TorchScript support `torch.device` function?

Reply:
I think it could but I am not totally sure about it. For `torch.device` we implemented a DeviceObjType in the type system and IValue for c10::Device